### PR TITLE
Support varargs in toHaveBeenCalledOnceWith (closes #517)

### LINKED ISSUE #517: toHaveBeenCalledOnceWith only asserts a match on the first parameter.
I'm not sure if this would be classed as a bug or a feature, so apologies for any mistakes raising the issue.

**Feature Request**

Description: toHaveBeenCalledOnceWith currently only matches against the first parameter provided to a mock function.

Possible solution:
In `toHaveBeenCalledOnceWith.js`, line 40 reads
`const pass = passOnce && this.equals(expected, received.mock.calls[0][0]);`
I believe that this should probably be changed to `this.equals(expected, received.mock.calls[0]);` and assert equality on the whole array of parameters passed to the mock function in the first call. This may need the function declaration to be changed to `toHaveBeenCalledOnceWith(received, ...expected)` or similar to handle multiple arguments.

If this is desired, or not so simple to implement, it would be good to make this clear in the documentation.

### PR DESCRIPTION
**Pull request by keeganwitt**

### CODE REVIEW

**Comment by keeganwitt** [comment]
I'm not sure this is the right way to solve this. It doesn't seem right...

File: src/matchers/toHaveBeenCalledOnceWith.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { isJestMockOrSpy } from '../utils';
 
-export function toHaveBeenCalledOnceWith(received, expected) {
+export function toHaveBeenCalledOnceWith(received, ...expected) {
```
Comment (Maxim-Mazurok) [inline src/matchers/toHaveBeenCalledOnceWith.js:3]: As [mentioned](https://github.com/jest-community/jest-extended/issues/518#issuecomment-1409835822) this PR should probably also change the types to match the new implementation

File: src/matchers/toHaveBeenCalledOnceWith.js [hunk 2]
```diff
@@ -16,7 +16,9 @@ export function toHaveBeenCalledOnceWith(received, expected) {
   }
 
   const passOnce = received.mock.calls.length === 1;
-  const pass = passOnce && this.equals(expected, received.mock.calls[0][0]);
+  const mockCall = received.mock.calls[0];
+  const receivedArray = Array.isArray(mockCall) ? mockCall : [mockCall];
```
Comment (Maxim-Mazurok) [inline src/matchers/toHaveBeenCalledOnceWith.js:20]: I have checked `createToBeCalledWithMatcher` implementation in jest:
image

And I've tried using `mockCall` directly without converting it to array and it seems to pass all your tests:

image

**Review by Maxim-Mazurok — CHANGES_REQUESTED** [review]

**Comment by Maxim-Mazurok** [comment]
> I'm not sure this is the right way to solve this. It doesn't seem right...

Not sure what exactly seems not right, perhaps my second suggestion will make it seem right? Haven't touched much of jest internals tho, so I probably don't know better.

File: src/matchers/toHaveBeenCalledOnceWith.js (hunk 2 above)
Comment (SimenB) [inline src/matchers/toHaveBeenCalledOnceWith.js:20]: It's always an array.

File: src/matchers/toHaveBeenCalledOnceWith.js (hunk 1 above)
Comment (keeganwitt) [reply src/matchers/toHaveBeenCalledOnceWith.js:3]: Yea, I realized I missed that at first. Thank you. Fixed.

File: src/matchers/toHaveBeenCalledOnceWith.js (hunk 2 above)
Comment (keeganwitt) [reply src/matchers/toHaveBeenCalledOnceWith.js]: Ah. I could have sworn I saw a case that wasn't, but I don't know what I thought I saw now 😛 
Removed the ternary.

**Comment by Maxim-Mazurok** [comment]
Thanks so much! Looking forward to getting this published to npm, cheers!

**Pull request merged**